src/main/java/org/testng/internal/Systematiser.java

```diff
@@ -0,0 +1,86 @@
+package org.testng.internal;
+
+import org.testng.ITestNGMethod;
+
+import java.util.Comparator;
+
+public class Systematiser {
+
+    private Systematiser() {
+        //Utility class. Defeat instantiation.
+    }
+
+    public static Comparator<Graph.Node> getComparator() {
```

Comment:
We cannot do this because the INSTANCES one doesn't work with a `ITestNGMethod` but merely works with just an `Object`. The `ITestNGMethod` based comparator also if you notice resorts to using the `ITestNGMethod` object **iff** the Node contains an `ITestNGMethod` object.